framework/include/utils/RankTwoTensor.h

```diff
@@ -315,29 +316,23 @@ class RankTwoTensorTempl : public TensorValue<T>
   /// returns _coords_ij * a_ij (sum on i, j)
   T doubleContraction(const RankTwoTensorTempl<T> & a) const;
 
-  /// returns C_ijkl = a_ij * b_kl
-  RankFourTensorTempl<T> outerProduct(const RankTwoTensorTempl<T> & a) const;
-
-  /// returns C_ijkl = a_ik * b_jl
-  RankFourTensorTempl<T> mixedProductIkJl(const RankTwoTensorTempl<T> & a) const;
-
-  /// returns C_ijkl = a_jk * b_il
-  RankFourTensorTempl<T> mixedProductJkIl(const RankTwoTensorTempl<T> & a) const;
-
-  /// returns C_ijkl = a_il * b_jk
-  RankFourTensorTempl<T> mixedProductIlJk(const RankTwoTensorTempl<T> & a) const;
+  /// returns C_ijkl = a_no * b_pq
+  template <int n, int o, int p, int q>
+  RankFourTensorTempl<T> times(const RankTwoTensorTempl<T> & b) const;
```

Comment:
The doxygen comment might need to be changed. While I know what you mean here, others reading this the first time may be confused, because it seems like we are creating indices i, j, k, l out of indices n, o, p, q, which is...magic.

This might be more accurate:
```
Assuming a bijective index map M: {n,o,p,q} -> {i,j,k,l}, this function returns C_{ijkl} = a_{M(n)M(o)} b_{M(p)M(q)}.
```

Reply:
This is great